core-bundle/src/Security/Voter/DataContainer/AbstractDynamicPtableVoter.php

```diff
@@ -78,16 +78,33 @@ private function getParentTableAndId(int $id): array
 
         // Limit to a nesting level of 10
         $records = $this->connection->fetchAllAssociative(
-            "SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=?".str_repeat(" UNION SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=@pid", 9),
-            [$id],
+            "SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=?".str_repeat(" UNION SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=@pid AND ptable=?", 9),
+            [$id, ...array_fill(0, 9, $table)],
```

Comment:
This should use named arguments, no?

Reply:
Not possible for table names. But the identifier should be quoted.

Reply:
Why does `?` work but `:table` wouldn't?

Reply:
Ah sorry, misunderstood your comment. Yes, that would work of course.

Reply:
changed in e102bfda6eeb09f9906f8bbcbe1653eb301dc507